### PR TITLE
client/asset: clarify eth and token versioning

### PR DESCRIPTION
**Pull request by buck54321**
~~This is needed for #2054.~~ The tokens supported versions were not exposed because tokens without wallets don't expose `WalletInfo` (no `Driver`).

Changing tacts on this a bit. Instead, we'll just have the tokens share the parent asset's versioning, and the server will maintain with a single protocol version. 

Just a reminder that server protocol version != contract version. A server's version does map to a contract version, but it's not one-to-one. i.e. you cannot add a new contract version without incrementing the server version, but a server version could be incremented for other reasons too. I've fixed versioning ambiguity in a couple places in this effort, but there is much more work done in #2038. I could potentially pull that out into a separate PR.

### CODE REVIEW

File: client/asset/eth/eth.go [hunk 1]
```diff
@@ -138,6 +138,8 @@ var (
 		},
 	}
 
+	SupportedTokenVersions = []uint32{0} // Server versions that we can handle.
```
[the remark this answers is not in the document]
Comment (buck54321) [reply client/asset/eth/eth.go:141]: Server protocol version != contract version. We've been sloppy with this concept before. The client knows how to handle a range of server versions. Those server versions certainly correspond to a range of contract versions, but not necessarily one-to-one.

File: client/asset/eth/eth.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ func registerToken(tokenID uint32, desc string, nets ...dex.Network) {
 	asset.RegisterToken(tokenID, token.Token, &asset.WalletDefinition{
 		Type:        "token",
 		Description: desc,
-	}, nets...)
+	}, SupportedTokenVersions, nets...)
```
[the remark this answers is not in the document]
Comment (buck54321) [reply client/asset/eth/eth.go:56]: Same answer here, really. From `NetTokens`, we would get the versioned contracts that have been deployed for a specific token. What we're interested in is which server protocol versions we can communicate with.

**Review by chappjc — APPROVED** [review]

File: client/asset/eth/eth.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (chappjc) [reply client/asset/eth/eth.go:141]: Ah right.  Say there's a new v1 swap contract version for USDC, server protocol version 1 (up from 0) could support contract versions 0 and 1.  Server advertises whichever single protocol version it supports, while client can manage multiple. Right, ok.  I think this works out but is definitely a bit hard to keep straight.

**Comment by chappjc** [comment]
Keeping open for discussion.  I think we're a little uncertain at present if the parent asset version is good for it's tokens or we need multiple sets.

File: client/asset/eth/eth.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (buck54321) [reply client/asset/eth/eth.go:141]: We've never really pinned down the versioning scheme. Lemme come up with a comprehensive plan and document it. This PR may not be the best way, now that I think about it.

**Review by martonp — APPROVED** [review]
I think it makes sense share the version for ETH and all tokens. I don't think a server would want to not support the latest of all contracts.

**Pull request merged**